### ISSUE TITLE
Unable to authenticate via nested LDAP groups

### ISSUE THREAD

**Issue by exodusftw** [propyless/panopuppet#47]
I had difficulty authenticating through nested ldap groups - I made the following changes to puppet/settings.py to resolve so I'm just asking for you to review and consider:

pupppet/settings.py

Line #93
Change: `from django_auth_ldap.config import LDAPSearch, ActiveDirectoryGroupType`
To: `from django_auth_ldap.config import LDAPSearch, NestedActiveDirectoryGroupType`

Line #103
Change: `ldap.SCOPE_SUBTREE, "(name=%(user)s)")`
To: `ldap.SCOPE_SUBTREE, "(sAMAccountName=%(user)s)")`

Line #106
Change: `AUTH_LDAP_GROUP_TYPE = ActiveDirectoryGroupType()`                               
To: `AUTH_LDAP_GROUP_TYPE = NestedActiveDirectoryGroupType()`

**Comment by propyless** [comment]
Hey there,

Is it compatible with non nested ldap groups?

**Comment by propyless** [comment]
I'm not so sure whether to or to not change this.. AD authentication can be a little bit complicated and i'm not sure I can get something universal to work...

What do you think about just documenting it?

**Comment by exodusftw** [comment]
I agree that it might not work if you force the change into master - I would be fine with simply documenting the change so people make configure as needed.

Ultimately - with a puppet class developed, templating the settings.py file to trigger off a parameter for nested AD groups should be fairly straightforward so these lines would be modified if something like a boolean is enabled for nested groups

**Comment by propyless** [comment]
Will document this soon

**Comment by propyless** [comment]
@exodusftw are you able to create a page in the wiki for this project?
If you are you could perhaps document it there if you have the time.
(if you don't I can ofcourse, but seeing as you have better knowledge over the problems you encountered)

**Comment by sheijmans** [comment]
`Is it compatible with non nested ldap groups?`
Just test this with a 2012R2 AD groups containing users and nested group.
It worked for both users (directly in the group and thru a nested group).

There a also 2 other Django ldap paramters to enable ldap group caching which might be  handy;
AUTH_LDAP_CACHE_GROUPS = True
AUTH_LDAP_GROUP_CACHE_TIMEOUT = 300

So perhaps this is merge-able, will update the ldap wiki with this info for now.
